### PR TITLE
Fix navigation bars padding and colors

### PR DESCRIPTION
**Pull request by ksharma-xyz**
### TL;DR

Improved edge-to-edge support and UI adjustments for better user experience, when 3 button nav bar is used.

The following bug https://issuetracker.google.com/issues/298296168 in edgetoedge api does not make the 3 button nav bar transparent when edgetoedge mode is enabled.

### What changed?

- Updated MainActivity to use `FLAG_LAYOUT_NO_LIMITS` for better edge-to-edge support or 3 button nav bar
- Refined padding and insets handling in SearchStopRow and SearchStopScreen
- Removed systemBarsPadding from TimeTableScreen

### Before

https://github.com/user-attachments/assets/b744b995-5aa0-4acb-9505-3c293c910976

### After

https://github.com/user-attachments/assets/c9f6976e-687d-43b9-bdd8-643437d1a717

### CODE REVIEW

**Comment by ksharma-xyz** [comment]
* **#198** <a href="https://app.graphite.dev/github/pr/ksharma-xyz/Krail/198?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ksharma-xyz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>
